core/src/main/java/com/google/instrumentation/stats/ContextUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import com.google.instrumentation.common.NonThrowingCloseable;
+import io.grpc.Context;
+
+/**
+ * Util methods/functionality to interact with the {@link io.grpc.Context}.
+ *
+ * <p>Users must interact with the current Context via the public APIs in {@link
+ * StatsContextFactory} and avoid usages of the {@link #STATS_CONTEXT_KEY} directly.
+ */
+public final class ContextUtils {
```

Comment:
I would suggest to keep it as ContextUtil because:
1) This is how we call it in the trace package as well.
2) Having "grpc" in the name of a class makes users think this is a gRPC specific library.
3) Advance users may need access to this key.